desktop/core/src/desktop/lib/fs/proxyfs.py

```diff
@@ -205,21 +206,25 @@ def restore(self, path):
   def create(self, path, *args, **kwargs):
     self._get_fs(path).create(path, *args, **kwargs)
 
+  def get_content_summary(self, path):
+    return self._get_fs(path).get_content_summary(path)
+
   def create_home_dir(self, home_path=None):
     """
-    Initially home_path will have path value for HDFS and if it is configured in Hue, try creating the user home dir for it first.
-    Then we check if S3/ABFS is configured in Hue via RAZ. If yes, try creating user home dir for them next.
+    Initially home_path will have path value for HDFS, try creating the user home dir for it first.
+    Then, we check if S3/ABFS is configured via RAZ. If yes, try creating user home dir for them next.
     """
     from desktop.conf import RAZ # Imported dynamically in order to have proper value.
 
-    if home_path is None:
-      home_path = self.get_home_dir()
-
```

Comment:
Do we not need this case at all anymore? When was this check actually needed?

Reply:
So this was present in that place from long-time but it was actually not called. `home_path` was never None.

Had it been then it would have crashed because ProxyFS was not having this method defined, so it was kind of like a ticking time-bomb error.

And we should not define it as well in ProxyFS because all other FS should handle that themselves for home_directory is incoming home_path is None design-wise. ProxyFS is just a wrapper connecting to other FS.